packages-internal/test-utils/src/mochaHooks.test.js

```diff
@@ -69,6 +69,7 @@ describe('mochaHooks', () => {
         let unsafeSetState;
         function Parent() {
           const [state, setState] = React.useState(0);
+          // TODO: uncomment once we enable eslint-plugin-react-compiler eslint-disable-next-line react-compiler/react-compiler -- unsafeSetState is required outside of component
```

Comment:
`unsafeSetState` is called outside of component
https://github.com/mui/material-ui/blob/79785b79514741486e5ab7d89003ed99ab3470c9/packages-internal/test-utils/src/mochaHooks.test.js#L84

Reply:
```suggestion
          // TODO: uncomment once we enable eslint-plugin-react-compiler // eslint-disable-next-line react-compiler/react-compiler -- unsafeSetState is required outside the component
```